src/sentry/static/sentry/app/views/performance/vitals-cards.tsx

```diff
@@ -0,0 +1,176 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import {Location} from 'history';
+
+import space from 'app/styles/space';
+import EventView from 'app/utils/discover/eventView';
+import {Organization} from 'app/types';
+import {Panel} from 'app/components/panels';
+import {formatPercentage} from 'app/utils/formatters';
+import VitalsCardsDiscoverQuery from 'app/views/performance/vitalDetail/vitalsCardsDiscoverQuery';
+import {getAggregateAlias, WebVital} from 'app/utils/discover/fields';
+import Link from 'app/components/links/link';
+import {t} from 'app/locale';
+
+import {
+  vitalAbbreviations,
+  vitalDetailRouteWithQuery,
+  vitalsBaseFields,
+  vitalsThresholdFields,
+} from './vitalDetail/utils';
+
+type Props = {
+  eventView: EventView;
+  organization: Organization;
+  location: Location;
+};
+
+export default function VitalsCards(props: Props) {
+  const {eventView, organization, location} = props;
+  const vitalsView = eventView.clone();
+
+  const shownVitals = [
+    WebVital.FP,
+    WebVital.FCP,
+    WebVital.LCP,
+    WebVital.FID,
+    WebVital.CLS,
+  ];
+
+  return (
+    <VitalsCardsDiscoverQuery
+      eventView={vitalsView}
+      orgSlug={organization.slug}
+      location={location}
+    >
+      {({isLoading, tableData}) => (
+        <React.Fragment>
+          <VitalsContainer>
+            {shownVitals.map(vitalName => (
+              <LinkedVitalsCard
+                key={vitalName}
+                vitalName={vitalName}
+                tableData={tableData}
+                isLoading={isLoading}
+                {...props}
+              />
+            ))}
+          </VitalsContainer>
+        </React.Fragment>
+      )}
+    </VitalsCardsDiscoverQuery>
+  );
+}
+
+const VitalsContainer = styled('div')`
+  display: grid;
+  grid-auto-flow: column;
+  gap: ${space(2)};
+`;
+
+type CardProps = Props & {
+  vitalName: WebVital;
+  tableData: any;
+  isLoading?: boolean;
+  noBorder?: boolean;
+};
+
+const NonPanel = styled('div')`
+  flex-grow: 1;
+`;
+
+const Card = styled(Panel)`
+  padding: ${space(2)};
+  cursor: pointer;
+`;
+
+const StyledQueryCard = styled(Card)`
+  &:focus,
+  &:hover {
+    top: -1px;
+    box-shadow: 0px 0px 0px 6px rgba(209, 202, 216, 0.2);
+    position: relative;
+    outline: none;
+  }
+`;
+
+export function LinkedVitalsCard(props: CardProps) {
+  const {vitalName} = props;
+  return (
+    <VitalLink {...props} vitalName={vitalName}>
+      <VitalsCard {...props} />
+    </VitalLink>
+  );
+}
+
+export function VitalsCard(props: CardProps) {
+  const {isLoading, tableData, vitalName, noBorder} = props;
+
+  const measurement = vitalAbbreviations[vitalName];
+
+  const Container = noBorder ? NonPanel : StyledQueryCard;
+
+  if (isLoading || !tableData || !tableData.data || !tableData.data[0]) {
```

Comment:
Optional chaining is great in these situations
```suggestion
  if (isLoading || !tableData?.data?.[0]) {
```